core/bin/migrations/20241024_clean_legacy_gcs.rs

```diff
@@ -0,0 +1,183 @@
+use anyhow::{anyhow, Context, Result};
+use dust::{
+    data_sources::{
+        data_source::{make_document_id_hash, DataSource},
+        file_storage_document::FileStorageDocument,
+    },
+    stores::{postgres, store},
+};
+use tokio_postgres::Row;
+
+use bb8::Pool;
+use bb8_postgres::PostgresConnectionManager;
+use futures::{StreamExt, TryStreamExt};
+use tokio_postgres::NoTls;
+use tokio_stream::{self as stream};
+
+async fn fetch_data_sources_batch(
+    pool: &Pool<PostgresConnectionManager<NoTls>>,
+    last_id: u64,
+    limit: usize,
+) -> Result<Vec<Row>, anyhow::Error> {
+    let c = pool.get().await?;
+
+    c.query(
+        "SELECT id, internal_id FROM data_sources WHERE id > $1 ORDER BY id ASC LIMIT $2",
+        &[&(last_id as i64), &(limit as i64)],
+    )
+    .await
+    .context("Query execution failed")
+}
+
+async fn clean_stored_versions_for_document_id(
+    pool: &Pool<PostgresConnectionManager<NoTls>>,
+    data_source: &DataSource,
+    data_source_id: i64,
+    document_id: &str,
+) -> Result<()> {
+    let c = pool.get().await?;
+
+    let document_versions = c.query("SELECT hash, created, status FROM data_sources_documents WHERE data_source = $1 AND document_id = $2", &[&data_source_id, &document_id]).await?;
```

Comment:
We can probably skip all documents created after July 25th to speed things up (see PR that removed double write [here](https://github.com/dust-tt/dust/pull/6405)).